### ISSUE TITLE
Cannot put metadata and doc to ResourceNameParameter

### ISSUE THREAD

**Issue by pshao25** [Azure/typespec-azure#809]
Here is how I define a resource before:
```
model DataManagerForAgriculture extends TrackedResourceBase {
  /**
   * DataManagerForAgriculture resource name.
   */
  @maxLength(63)
  @minLength(1)
  @pattern("^[A-Za-z0-9]+(-[A-Za-z0-9]+)*$")
  @path
  @key("dataManagerForAgricultureResourceName")
  @segment("farmBeats")
  @visibility("read")
  name: string;
}
```
I have customized document. I have length limitation by `@maxLength` and `@minLength`.

After I change it to
```
model DataManagerForAgriculture extends Foundations.TrackedResource {
  /**
   * DataManagerForAgriculture resource name.
   */
  ...ResourceNameParameter<
    DataManagerForAgriculture,
    "dataManagerForAgricultureResourceName",
    "farmBeats",
    "^[A-Za-z0-9]+(-[A-Za-z0-9]+)*$"
  >;
}
```
The doc I added is ignored. And I cannot add more decorators to the spread `ResourceNameParameter`.

**Comment by markcowl** [comment]
You can do this using an augment decorator on the name parameter, [as in this playground](https://azure.github.io/typespec-azure/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3Jl3yvIK3Jlc291cmNlLW1hbmFnZXIiOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TRFVbpAKnIG0HEZS5Db3JlzhJSx3dNxnY7CgovKiogQ29udG9zbyDIHiBQcm92aWRlciDmAJ5tZW50IEFQSS4gKi8KQGFybcggTmFtZXNwYWNlCkBzZXJ2aWNlKHsKICB0aXRsZTogIsdYyC5IdWJDbGllbnQiLAp9KQpA5wFgZWQo5wDBcykKbshSIE1pY3Jvc29mdC7SR%2B8AuEFQSSDHTXPkAKNlbnVtIMhUIOQAksQuMjAyMS0xMC0wMS1wcmV2aWV3yDXENCAgQHVzZURlcGVuZGVuY3ko9QEx6AFrcy52MV8wX1DGSF8xKcRAYXJtQ29tbW9u5AGZc8cq10jLKctUNcRIYPIAqWAsCn3mAPNB6AD16wEOIOgCTeQAyG1vZGVsIEVtcGxveWVlIGlzIFRyYWNrZWToAII8yBxQcm9wZXJ0aWVzPuUBJC4u6QCm5AHYUGFyYW1ldGVyyTE%2BO%2BQAhkBAZG9jKMgULuQBsSwgIlRoaXPEbWhvdyB5b3Ugb3ZlcnJpZGUgdGhlIGRvY3MiKecBtcg6IHDpAIDyALjqAJzpAb9BZ2Ugb2YgZcg%2F5QG0YWdlPzogaW50MzI7CscpQ2l0edIqY2l0eT86IHN0cuUDJ8csUHJvZmls01lAZW5jb2RlKCJiYXNlNjR1cmwi5AGccMYwPzogYnl0ZXPJSFRoZSBzdGF0dXPES%2BQA9Gxhc3Qg5ADFYXRpb27lAzogIEB2aXNpYmlsaXR5KCJyZWFkyFXEGOUDwVN0YXTEX%2BUB38wU5QF8xGtBbsRXbiB15AK5d2l0aCBvbmUgY2hvaeYCAXVuxBlTaW5nbGVWYWx1ZVXFEegBR3RoZXLlAhdvbmx5zzsgIE9uZUNIxA86ICLEDMUbIizkAOfmATrpAonkAPLsALnlAP%2FlAUlh6QKQ5gQwbHJv5ADCdXPnAJvxANvlAJzIX%2BsBUchHIGNyZcQncmVxdWVzdCBoYXMgYmVlbiBhY2NlcHRlZMRnICBBxw46ICLIC%2BYAutFQacRA5AC06QDByETsAJw6ICLMD9pMdXBkYXTET8VDVccOOiAiyAvKO%2BkFYukAxOYA3GTnAiFTdWNjZWVk5QDFyQzTP8U25AFNZmFpbMk%2BRsUNOiAixgncOHdhcyBjYW5jZco%2BQ8cP5AW5xwv%2FAUAgZGVsZekBgETEDeYA%2BcgL5AXp7wP2bW926gHK6QP4TW92ZVLHFegCqsRzbW92xGhmcm9tIGxvY%2BYAvMVuxBPxA8%2FLM3RvzzF0b8ov9wCVc3BvbnPrBUbmAJbHFuwAl%2B4D68U%2BxWTGfOYC7s1uaW50ZXJm5AbXT%2BgEE3MgZXh0ZW5kc%2FYHmy7LKXvkBXdhcm3II8oby1noAM3mBTJnZXTkAadByjVSZWFk7AXLIOcCe09y5QKn5QHWyy9Dxx1SZXBsYWNlQXN5bmPOP%2BUC99A3UGF0Y2hTzCws8wXVxUDmAj3PQOUCOmVXaXRob3V0T2vTd2xpc3RCecgwR3JvdXDPREzFIlBhcmVudNQ8U3Vic2NyaXDlAhzGO8YzzBnMOegGgiBzYW1wbOsDCWFjxUR0aGF05gIG6QZFdG8gZGlmZmXkAITvAonFKe4AskHFSO8BN%2BsDFMgN5gKL8wCSSEVBROoGbsR%2BY2hlY2vqAKpleGlzdGVu6AXoxR5FyRTvAfvNHe4H%2Fw%3D%3D&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-resource-manager%2Fall%22%5D%7D%7D)

**Comment by pshao25** [comment]
Thanks! Seems the link is wrong, but I already got your point! Augment decorator works!

My follow-up problem is the type of `name` in `ResourceNameParameter` is hard coded as `string`. What if the name is an enum? For this [swagger](https://github.com/Azure/azure-rest-api-specs/blob/5a281cf0d538de6dad0c70eda7ee901c60a11e6b/specification/playwrighttesting/resource-manager/Microsoft.AzurePlaywrightService/preview/2023-10-01-preview/playwrighttesting.json#L195-L215), we previously defined its TypeSpec as 
```
model Quota is ProxyResource<QuotaProperties> {
  /**
   * The quota name.
   */
  @path
  @key("name")
  @segment("quotas")
  @visibility("read")
  name: QuotaNames;
}
```
After I upgraded to the latest compiler, a non-string name is not allowed.
